Makefile

```
@@ -0,0 +1,57 @@
+PROJECT := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+SRC := $(PROJECT)src
+TESTS := $(PROJECT)tests
+ALL := $(SRC) $(TESTS)
+
+export PYTHONPATH = $(PROJECT):$(PROJECT)/lib:$(SRC)
+
+update-dependencies:
+	uv lock -U --no-cache
+
+generate-requirements:
+	uv pip compile -q --no-cache pyproject.toml -o requirements.txt
+	uv pip compile -q --no-cache --all-extras pyproject.toml -o requirements-dev.txt
+
+lint:
+	uv tool run ruff check $(ALL)
+	uv tool run ruff format --check --diff $(ALL)
+
+fmt:
+	uv tool run ruff check --fix-only $(ALL)
+	uv tool run ruff format $(ALL)
```

`uv tool run` does not respect your pyproject.toml or uv.lock files but instead uses the latest tool available.  As is, the `ruff` version called out in pyproject.toml here is meaningless.  To pin a linting tool here, you'd have to hard code it into these commands (`uv tool run ruff@0.7.3` or `uv tool run ruff --from "ruff=0.7.3"`)

An alternative is using `uv run --isolated`, which makes a one-off isolated environment, eg:
```suggestion

lint: EXTRA_DEPENDENCIES = --extra dev
lint:
	uv run --isolated $(EXTRA_DEPENDENCIES) ruff check $(ALL)
	uv run --isolated $(EXTRA_DEPENDENCIES) ruff format --check --diff $(ALL)

fmt: EXTRA_DEPENDENCIES = --extra dev
fmt:
	uv run --isolated $(EXTRA_DEPENDENCIES) ruff check --fix-only $(ALL)
	uv run --isolated $(EXTRA_DEPENDENCIES) ruff format $(ALL)
```

The `EXTRA_DEPENDENCIES = ...` is in there because it was the least objectionable way I saw for handling multiple optional dependency groups (I needed at least 2, because I didn't want pytest operator running during unit tests).  But if having a single group is fine, it can be removed

---

in our case we don't have pytest operator or any 'heavy' dependency in integration tests, so we should be good